mock/py/mockbuild/podman.py

```diff
@@ -52,35 +35,37 @@ def install_pkgmgmt_packages(self):
         self.exec(cmd)
 
     @traceLog()
-    def export(self, cache_file_name, compress_program):
-        """ export container and compress it  """
-        getLog().info("Exporting container: %s as %s", self.image, cache_file_name)
-        cmd_podman = ["podman", "export", self.container_id]
-        podman = subprocess.Popen(cmd_podman, stdout=subprocess.PIPE)
-        cache_file = open(cache_file_name, "w")
-        cmd_compressor = [compress_program, "--stdout"]
-        compressor = subprocess.Popen(cmd_compressor, stdin=podman.stdout, stdout=cache_file)
-        compressor.communicate()
-        podman.communicate()
-        cache_file.close()
+    def exec(self, command):
+        """ exec command in container """
+        util.do(command, chrootPath=self.buildroot.make_chroot_path(),
+                printOutput=True, env=self.buildroot.env)
+
+    @contextmanager
+    def mounted_image(self):
+        cmd_mount = ["podman", "image", "mount", self.image]
+        cmd_umount = ["podman", "image", "umount", self.image]
+        output = subprocess.check_output(cmd_mount)
+        mountpoint = output.decode("utf-8").strip()
+        getLog().info("mounting %s with podman image mount", self.image)
+        try:
+            getLog().info("image %s as %s", self.image, mountpoint)
+            yield mountpoint
+        finally:
+            getLog().info("umounting image %s (%s) with podman image umount",
+                          self.image, mountpoint)
+            output = subprocess.check_output(cmd_umount)
 
     @traceLog()
     def cp(self, destination, tar_cmd):
         """ copy content of container to destination directory """
         getLog().info("Copy content of container %s to %s", self.image, destination)
-        cmd_podman = ["podman", "export", self.container_id]
-        podman = subprocess.Popen(cmd_podman, stdout=subprocess.PIPE)
-        cmd_tar = [tar_cmd, "-xC", destination, "-f", "-"]
-        tar = subprocess.Popen(cmd_tar, stdin=podman.stdout)
-        tar.communicate()
-        podman.communicate()
-
-    @traceLog()
-    def remove(self):
-        """ remove the container """
-        cmd = ["podman", "rm", "-f", self.container_id]
-        util.do(cmd)
-        self.container_id = None
+        with self.mounted_image() as mount_path:
+            cmd_podman = [tar_cmd, "-C", mount_path, "-c", "."]
```

Comment:
I would put a comment above this line. When you replaced "podman export" with tar_cmd it is less obvious what you are doing here. 
```
# create a tarball from mounted podman image and send it to PIPE
```